lms/static/js/ccx/view/ccx_schedule.js

```diff
@@ -0,0 +1,394 @@
+        );
+        this.unSavedChanger = true;
+        this.collection.reset(this.scheduleJson);
+        this.render();
+      },
+
+      showUnitInScheduleTree: function(
+        chapterLocation, sequentialLocation, verticalLocation,
+        startDate, dueDate
+      ) {
+        // hide child (can be chapter, sequential or vertical) in collection.
+        var units = this.findLineage(
+          this.scheduleJson,
+          chapterLocation,
+          sequentialLocation,
+          verticalLocation
+        );
+
+        if (units) {
+          this.errorMessage = this.validUnitDates(startDate, dueDate);
+          if (!this.errorMessage) {
+            units.map(this.show);
+            var unit = units[units.length - 1];
+            if (!_.isUndefined(unit)) {
+              unit.start = startDate;
+              unit.due = dueDate;
+            }
+            this.changeVisibilityOfUnitInSchedule([unit], this.show);
+            this.unSavedChanger = true;
+            this.collection.reset(this.scheduleJson);
+            this.render();
+          } else {
+            this.showErrorMessage();
+          }
+        }
+      },
+
+      showAllUnitsInScheduleTree: function() {
+        // show all units i.e chapters, subsections and verticals
+        // in schedule tree
+        this.changeVisibilityOfUnitInSchedule(
+          this.scheduleJson, this.show
+        );
+        this.unSavedChanger = true;
+        this.collection.reset(this.scheduleJson);
+        this.render();
+      },
+
+      applyUnitToTree: function(tree, unit, chapterLocation, sequentialLocation) {
+        // sync tree ith unit
+        _.each(tree, function(chapter, chapterIndex) {
+          if (chapterLocation) {
+            if (chapter.location ===  chapterLocation) {
+              if (chapter.location ===  unit.location) {
+                tree[chapterIndex] = unit;
+                return true;
+              }
+              _.each(chapter.children, function(subSection, subSectionIndex) {
+                if (subSection.location ===  sequentialLocation) {
+                  if (subSection.location ===  unit.location) {
+                    // update that subsection in tree
+                    tree[chapterIndex].children[subSectionIndex] = unit;
+
+                    _.each(subSection.children, function(__, verticalIndex) {
+                      // update start and due dates of verticals
+                      tree[chapterIndex].children[subSectionIndex].
+                        children[verticalIndex].start = unit.start;
+
+                      tree[chapterIndex].children[subSectionIndex].
+                        children[verticalIndex].due = unit.due;
+                    });
+                  }
+                  return true;
+                }
+              });
+              return true;
+            }
+          }
+        });
+      },
+
+      hideUnitOnTree: function(tree, unit, chapterLocation, sequentialLocation) {
```

Comment:
This and the code above look near identical. can you refactor them into one?

Also, given that this is centric around the data, can you push this into the collection/model as appropriate?